syntax = "proto3";

import "types.proto";

// Categories - сервис по работе с категориями.
service Categories {
    // PredictCategories - предсказать категории по тексту статей.
    rpc PredictCategories(PredictCategoriesRequest) returns(PredictCategoriesResponse);
}

// PredictCategoriesRequest - запрос на предсказание категорий по тексту статьи.
message PredictCategoriesRequest {
    // text - текст статьи.
    string text = 1;
}

// PredictCategoriesResponse - ответ на PredictCategoriesRequest.
message PredictCategoriesResponse {
    // result - результат.
    repeated CategoryPrediction result = 1;
}